prql-compiler/src/semantic/resolver.rs

```diff
@@ -552,31 +554,72 @@ impl Resolver {
 
         let func_name = &closure.name;
 
-        let (tables, other): (Vec<_>, Vec<_>) = zip(&closure.params, to_resolve.args)
+        let (tables, other): (Vec<_>, Vec<_>) = dbg!(zip(&closure.params, to_resolve.args)
             .enumerate()
             .partition(|(_, (param, _))| {
-                let is_table = param
+                param
                     .ty
                     .as_ref()
                     .map(|t| matches!(t, Ty::Table(_)))
-                    .unwrap_or_default();
-
-                is_table
-            });
+                    .unwrap_or_default()
+            }));
 
-        let has_tables = !tables.is_empty();
+        let has_tables = !dbg!(&tables).is_empty();
```

Comment:
This "has_tables" is badly named. It should be "has_relations" and it is true for all functions that take a relation as an arguments: `from`, `select`, `take`...

Code here is injecting `_frame` and `_right` modules into the context, but only for the duration of resolving arguments of this functions call. In effect, this allows you this:

```
FunctionCall:
  args:
  - column foo   <- this foo can "see" relation a and resolve to a.foo
  - relation a
```

... so this is not where you'd want to infer modules in `default_db`.

Reply:
OK, I commented out this code

Reply:
The reason I was thinking to add it here is that presumably we only want to add modules when the reference is somewhere we could expect relations. So for example:

```
from x.y.z # we want to add an x module here
select a.b.c # wee don't want to add an a or b module here
```

But probably we could do that elsewhere?

Reply:
Oh, I see the reasoning.

Yes, we could handle this elsewhere, but I don't know where exactly at the moment. Will come back to this.

Reply:
My investigation into this issue ended in major change with how declarations are inferred. So I opened a PR that implements this.